### PR TITLE
[SIM_FLUTTER-46] [INTEGRATE] Create and implement admin delete user

### PR DESCRIPTION
**Pull request by egiegarciano14**
### Issue links
- [Backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-46)
- [Slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1688378714713739)

### Definition of Done
- [x] Admin can delete user

### Note
- Dependent to [pr#39](https://github.com/framgia/sph-flutter/pull/39)

### Recording
image

### CODE REVIEW

File: frontend/lib/src/services/user_service.dart [hunk 1]
```diff
@@ -17,4 +20,31 @@ class UserService {
       return [];
     }
   }
+
+  static Future<dynamic> deleteUser(String id) async {
```
Comment (jstephend-sun) [inline frontend/lib/src/services/user_service.dart]:
```suggestion
  static Future<void> deleteUser(String id) async {
```
function doesnt have any return

File: frontend/lib/src/controllers/admin_user_list_controller.dart [hunk 2]
```diff
@@ -12,9 +12,13 @@ class AdminUserListController extends GetxController {
...
 
-  Future<List<User>> getUsers() async {
-    final result = await UserService.getUsers();
+  Future<List<User>> getUsers({String keyword = ''}) async {
+    final result = await UserService.getUsers(keyword: keyword);
     users.assignAll(result);
     return result;
   }
+
+  Future<dynamic> deleteUser(String id) async {
```
Comment (jstephend-sun) [inline frontend/lib/src/controllers/admin_user_list_controller.dart]:
```suggestion
  Future<void> deleteUser(String id) async {
```
no return

File: frontend/lib/src/features/admin_user_list/components/user_list_tile.dart [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
 import 'package:frontend/src/features/admin_user_list/components/user_delete_dialog.dart';
+import 'package:frontend/src/models/user.dart';
 
 /*
   ListTile that is used for Admin User List Page
 
-  @param name, full name of the user to be displayed
-  
-  @param isAdmin, bool that controls prevention of user deletion (cant delete admin user)
+  @param users: contains user details.
```
Comment (jstephend-sun) [inline frontend/lib/src/features/admin_user_list/components/user_list_tile.dart]:
```suggestion
  @param user: contains user details.
```

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changed #42

review time: 10mins
comments: 3

**Review by jstephend-sun — APPROVED** [review]
Approved 

review time: 10mins
comments: 3
iteration: 1st review

**Review by Dbulawan13 — APPROVED** [review]
Approve PR #42 

Working:
- Able to delete user
- Clicking (X) or No button will close the modal
- Unable to delete admin accounts
- Toast message will appear after deleting a user account

https://github.com/framgia/sph-flutter/assets/86758474/a244120a-9373-4eb4-94b2-5a0ac0e33fd3

Testing Hours: 0.16
Comment: 0
Iteration: 0

**Pull request merged**